diff -ru toybox-0.5.2.orig/toys/posix/grep.c toybox-0.5.2/toys/posix/grep.c
--- toybox-0.5.2.orig/toys/posix/grep.c	2015-02-26 02:42:24.000000000 +0000
+++ toybox-0.5.2/toys/posix/grep.c	2015-03-13 20:07:48.017855796 +0000
@@ -76,6 +76,17 @@
     return;
   }
 
+  struct stat stat;
+  if (fstat(fd, &stat) != 0) {
+    perror_msg("%s", name);
+    fclose(file);
+    return;
+  }
+  if (S_ISDIR(stat.st_mode)) {
+    fclose(file);
+    return;
+  }
+  
   // Loop through lines of input
   for (;;) {
     char *line = 0, *start;
