--- a/include/istream	2015-07-18 08:23:44.294261748 +0100
+++ b/include/istream	2015-07-18 08:21:21.092822795 +0100
@@ -69,6 +69,8 @@
 		basic_istream<charT,traits>& operator>>(unsigned int& n);
 		basic_istream<charT,traits>& operator>>(long& n);
 		basic_istream<charT,traits>& operator>>(unsigned long& n);
+		basic_istream<charT,traits>& operator>>(unsigned long long int& n);
+		basic_istream<charT,traits>& operator>>(long long int& n);
 		basic_istream<charT,traits>& operator>>(void*& p);
 		basic_istream<charT,traits>& operator>>(basic_streambuf<char_type,traits>* sb);
 
@@ -183,6 +185,7 @@
 				}
 				s[i] = c;
 			}
+			count_last_ufmt_input = n-1;
 			s[n-1] = traits::eos();
 			return *this;
 		}
@@ -447,6 +450,22 @@
 		return *this;
 	}
 
+	template <class charT, class traits> _UCXXEXPORT basic_istream<charT,traits>&
+		basic_istream<charT,traits>::operator>>(unsigned long long int& n)
+	{
+		sentry(*this);
+		__istream_readin<traits, charT, unsigned long long int>::readin(*this, n);
+		return *this;
+	}
+
+	template <class charT, class traits> _UCXXEXPORT basic_istream<charT,traits>&
+		basic_istream<charT,traits>::operator>>(long long int& n)
+	{
+		sentry(*this);
+		__istream_readin<traits, charT, long long int>::readin(*this, n);
+		return *this;
+	}
+
 #ifdef __UCLIBCXX_HAS_FLOATS__
 	template <class charT, class traits> _UCXXEXPORT basic_istream<charT,traits>&
 		basic_istream<charT,traits>::operator>>(float& n)
--- a/include/istream_helpers.orig	2012-05-25 22:15:31.000000000 +0100
+++ b/include/istream_helpers	2015-07-17 20:33:33.170097338 +0100
@@ -300,6 +300,57 @@
 		}
 	};
 
+	template <class traits> class _UCXXEXPORT __istream_readin<traits, char, unsigned long long int>{
+	public:
+		inline static void readin(basic_istream<char, traits >& stream, unsigned long long int & var)
+		{
+			basic_string<char, traits > temp;
+
+			if(stream.flags() & ios_base::dec){
+				temp = _readTokenDecimal( stream);
+				sscanf(temp.c_str(), "%llu", &var );
+			}else{
+				temp = _readToken( stream);
+				if( stream.flags() & ios_base::oct){
+					sscanf(temp.c_str(), "%llo", &var );
+				}else if(stream.flags() & ios_base::hex){
+					if(stream.flags() & ios_base::uppercase){
+						sscanf(temp.c_str(), "%llX", &var );
+					}else{
+						sscanf(temp.c_str(), "%llx", &var);
+					}
+				}else{
+					sscanf(temp.c_str(), "%lli", (long long int *)(&var) );
+				}
+			}
+		}
+	};
+
+	template <class traits> class _UCXXEXPORT __istream_readin<traits, char, long long int>{
+	public:
+		inline static void readin(basic_istream<char, traits >& stream, long long int & var)
+		{
+			basic_string<char, traits > temp;
+
+			if(stream.flags() & ios_base::dec){
+				temp = _readTokenDecimal( stream);
+				sscanf(temp.c_str(), "%lld", &var );
+			}else{
+				temp = _readToken( stream);
+				if( stream.flags() & ios_base::oct){
+					sscanf(temp.c_str(), "%llo", &var );
+				}else if(stream.flags() & ios_base::hex){
+					if(stream.flags() & ios_base::uppercase){
+						sscanf(temp.c_str(), "%llX", &var );
+					}else{
+						sscanf(temp.c_str(), "%llx", &var);
+					}
+				}else{
+					sscanf(temp.c_str(), "%lli", (long long int *)(&var) );
+				}
+			}
+		}
+	};
 
 #ifdef __UCLIBCXX_HAS_FLOATS__
 
--- a/src/istream.cpp.orig	2012-05-25 22:15:31.000000000 +0100
+++ b/src/istream.cpp	2015-07-17 20:32:14.090381350 +0100
@@ -56,6 +56,8 @@
 	template _UCXXEXPORT istream & istream::operator>>(unsigned int &n);
 	template _UCXXEXPORT istream & istream::operator>>(long unsigned &n);
 	template _UCXXEXPORT istream & istream::operator>>(long int &n);
+	template _UCXXEXPORT istream & istream::operator>>(unsigned long long int &n);
+	template _UCXXEXPORT istream & istream::operator>>(long long int &n);
 	template _UCXXEXPORT istream & istream::operator>>(void *& p);
 	template _UCXXEXPORT istream & operator>>(istream & is, char & c);
 
