diff -ru toybox-0.5.1/toys/posix/find.c toybox-0.5.1.new/toys/posix/find.c
--- toybox-0.5.1/toys/posix/find.c	2014-11-20 03:38:00.000000000 +0000
+++ toybox-0.5.1.new/toys/posix/find.c	2015-01-26 21:50:14.627256562 +0000
@@ -430,7 +430,7 @@
           if (aa->dir && TT.topdir == -1) TT.topdir = xopen(".", 0);
 
         // collect names and execute commands
-        } else if (check) {
+        } else {
           char *name, *ss1 = ss[1];
           struct double_list **ddl;
 
@@ -438,6 +438,7 @@
           aa = (void *)llist_pop(&argdata);
           ss += aa->arglen + 1;
 
+          if (check) {
           // name is always a new malloc, so we can always free it.
           name = aa->dir ? xstrdup(new->name) : dirtree_path(new, 0);
 
@@ -474,6 +475,7 @@
           dlist_add(ddl, name);
           aa->namecount++;
           if (!aa->plus) test = flush_exec(new, aa);
+          }
         }
 
         // Argument consumed, skip the check.
