diff -ru toybox-0.5.2.orig/toys/posix/cp.c toybox-0.5.2/toys/posix/cp.c
--- toybox-0.5.2.orig/toys/posix/cp.c	2015-02-26 02:42:24.000000000 +0000
+++ toybox-0.5.2/toys/posix/cp.c	2015-03-13 20:08:18.990360352 +0000
@@ -391,6 +391,7 @@
   }
 
   if (toys.optflags & FLAG_D) {
+    TT.destname = toys.optargs[toys.optc-1];
     if (mkpathat(AT_FDCWD, TT.destname, 0, 2))
       perror_exit("-D '%s'", TT.destname);
     if (toys.optc == 1) return;
