diff -ru toybox-0.5.1/lib/portability.c toybox-0.5.1.new/lib/portability.c
--- toybox-0.5.1/lib/portability.c	2014-11-20 03:38:00.000000000 +0000
+++ toybox-0.5.1.new/lib/portability.c	2015-01-29 09:46:46.781228441 +0000
@@ -37,7 +37,7 @@
       new_line = realloc(*linep, new_len);
       if (!new_line) return -1;
       *np = new_len;
-      *linep = new_line;
+      line = *linep = new_line;
     }
 
     line[i] = ch;
@@ -51,7 +51,7 @@
     new_line = realloc(*linep, new_len);
     if (!new_line) return -1;
     *np = new_len;
-    *linep = new_line;
+    line = *linep = new_line;
   }
   line[i + 1] = '\0';
 
diff -ru toybox-0.5.1/toys/posix/grep.c toybox-0.5.1.new/toys/posix/grep.c
--- toybox-0.5.1/toys/posix/grep.c	2014-11-20 03:38:00.000000000 +0000
+++ toybox-0.5.1.new/toys/posix/grep.c	2015-01-29 09:45:41.249231496 +0000
@@ -65,6 +65,17 @@
     return;
   }
 
+  struct stat stat;
+  if (fstat(fd, &stat) != 0) {
+    perror_msg("%s", name);
+    fclose(file);
+    return;
+  }
+  if (S_ISDIR(stat.st_mode)) {
+    fclose(file);
+    return;
+  }
+  
   for (;;) {
     char *line = 0, *start;
     regmatch_t matches[3];
